Userland/Utilities/pkg/AvailablePort.cpp

```diff
@@ -0,0 +1,176 @@
+    if (!possible_available_port.has_value()) {
+        outln("pkg: No match for queried name \"{}\"", package_name);
+        return 0;
+    }
+
+    auto& available_port = possible_available_port.release_value();
+
+    outln("{}: {}, {}", available_port.name(), available_port.version(), available_port.website());
+    if (verbose) {
+        out("Installed: ");
+        if (is_installed(installed_ports, package_name))
+            outln("Yes");
+        else
+            outln("No");
+    }
+    return 0;
+}
+
+static Optional<Markdown::Table::Column const&> get_column_in_table(Markdown::Table const& ports_table, StringView column_name)
+{
+    for (auto& column : ports_table.columns()) {
+        if (column_name == column.header.render_for_terminal())
+            return column;
+    }
+    return {};
+}
+
+ErrorOr<int> AvailablePort::update_available_ports_list_file()
+{
+    if (!Core::System::access("/usr/Ports/AvailablePorts.md"sv, R_OK).is_error() && FileSystem::remove("/usr/Ports/AvailablePorts.md"sv, FileSystem::RecursionMode::Disallowed).is_error()) {
+        outln("pkg: /usr/Ports/AvailablePorts.md exists, but can't delete it before updating it!");
+        return 0;
+    }
+    RefPtr<Protocol::Request> request;
+    auto protocol_client = TRY(Protocol::RequestClient::try_create());
+    (void)protocol_client;
+    HashMap<DeprecatedString, DeprecatedString, CaseInsensitiveStringTraits> request_headers;
+    Core::ProxyData proxy_data {};
+
+    auto output_stream = TRY(Core::File::open("/usr/Ports/AvailablePorts.md"sv, Core::File::OpenMode::ReadWrite, 0644));
+    Core::EventLoop loop;
+
+    URL url("https://raw.githubusercontent.com/SerenityOS/serenity/master/Ports/AvailablePorts.md");
+    DeprecatedString method = "GET";
+    outln("pkg: Syncing packages database...");
+    request = protocol_client->start_request(method, url, request_headers, ReadonlyBytes {}, proxy_data);
+    request->on_finish = [&](bool success, auto) {
+        if (!success)
+            outln("pkg: Syncing packages database failed.");
+        else
+            outln("pkg: Syncing packages database done.");
+        loop.quit(success ? 0 : 1);
+    };
+    request->stream_into(*output_stream);
+    return loop.exec();
+}
+
+ErrorOr<HashMap<String, AvailablePort>> AvailablePort::read_available_ports_list()
+{
+    auto available_ports_file = TRY(Core::File::open("/usr/Ports/AvailablePorts.md"sv, Core::File::OpenMode::Read, 0600));
+    auto content_buffer = TRY(available_ports_file->read_until_eof());
+    auto content = StringView(content_buffer);
+    auto document = Markdown::Document::parse(content);
+    auto finder = MarkdownTableFinder::analyze(*document);
+    if (finder.table_count() != 1)
+        return Error::from_string_literal("Invalid tables count in /usr/Ports/AvailablePorts.md");
+
+    VERIFY(finder.tables()[0]);
+    auto possible_port_name_column_index = get_column_in_table(*finder.tables()[0], "Port"sv);
```

Comment:
Why are these called indexes when they are actually of the type `Markdown::Table::Column&` ? If it was actually an index, I would expect the type to be size_t (or `Markdown::Something::FooIndex`)